.github/workflows/pypi-publish.yml

```diff
@@ -1,31 +1,92 @@
-name: PyPI Publish
+name: Build and upload python wheels
```

Comment:
I've reverted this change so the workflow will still be the same.  Do GA names matter for workflows?  Or jobs?  The job name has to change because its splitting one job into three.

Reply:
I think workflows. I'm mostly thinking of this left sidebar on https://github.com/lancedb/lancedb/actions

There some some dead names there, which I find annoying.